diff --git a/Lib/tarfile.py b/Lib/tarfile.py
index 6e092f1dcee5e8..a4f9ce3311f6da 100644
--- a/Lib/tarfile.py
+++ b/Lib/tarfile.py
@@ -2841,9 +2841,11 @@ def makelink_with_filter(self, tarinfo, targetpath,
                     "makelink_with_filter: if filter_function is not None, "
                     + "extraction_root must also not be None")
             try:
-                filter_function(
+                filtered = filter_function(
                     unfiltered.replace(name=tarinfo.name, deep=False),
                     extraction_root)
+                if filtered is None:
+                    return
                 filtered = filter_function(unfiltered, extraction_root)
             except _FILTER_ERRORS as cause:
                 raise LinkFallbackError(tarinfo, unfiltered.name) from cause
